lib/validation/volumes.js

```diff
@@ -51,6 +54,39 @@ function validateVolumeName(volumeName, opts) {
     return err;
 }
 
+// NOTE: This duplicates code from sdc-volapi's lib/validation/volumes.js and
+//       should probably eventually be put in a library shared between the two.
+function validateVolumeNameSearchParam(name) {
+    var coreName;
+    var err;
+    var nameBegin;
+    var nameLen;
+
+    // 'name' is special because we allow '*' as a prefix or suffix for
+    // wildcard searching. As such, we need to validate that the name is
+    // valid whether or not it is surrounded by '*'s.
+    nameBegin = 0;
+    nameLen = name.length;
```

Comment:
Should we validate first that `name` is a string with something like `if (typeof (name) !== 'string') { err = new Error(...); return err; }`? Otherwise it seems that this function could crash when `name` is not a string.

This function is only used to validate the query parameters of a GET request, and thus it seems it wouldn't be called with a non-string value for now, but if we start using it to validate e.g POST requests' JSON data, we might run into this.